controllers/metering/controllers/metering_controller.go

```diff
@@ -20,210 +20,238 @@ import (
+		for resourceName, resourceInfo := range resources.Spec.Resources {
+			if err := r.Get(ctx, types.NamespacedName{Namespace: r.MeteringSystemNameSpace, Name: MeteringPrefix + resourceInfo.NameSpace}, &metering); err != nil {
+				r.Logger.Error(err, "get metering failed", "name", MeteringPrefix+resourceInfo.NameSpace)
+				continue
+			}
+			if _, ok := metering.Spec.Resources[resourceName]; !ok {
+				r.Logger.Error(err, "resource not found in metering", "name", resourceName)
+				continue
+			}
+			if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, &metering, func() error {
+				metering.Spec.Resources[resourceName].Used.Add(*resourceInfo.Used)
+				return r.Update(ctx, &metering)
+			}); err != nil {
+				return ctrl.Result{Requeue: true}, err
+			}
+
+			resources.Status.Status = meteringv1.Complete
+			err := r.Update(ctx, resources)
+			if err != nil {
+				return ctrl.Result{Requeue: true}, err
+			}
 		}
-		//// metering finalizer
-		//if metering.DeletionTimestamp.IsZero() {
-		//	if controllerutil.AddFinalizer(&metering, FinalizerName) == true {
-		//		//TODO delete resource quota
-		//		if err := r.Update(ctx, &metering); err != nil {
-		//			return ctrl.Result{}, err
-		//		}
-		//	}
-		//} else {
-		//	if controllerutil.RemoveFinalizer(&metering, FinalizerName) == true {
-		//		if err := r.Update(ctx, &metering); err != nil {
-		//			return ctrl.Result{}, err
-		//		}
-		//	}
-		//	return ctrl.Result{}, nil
-		//}
-	}
-	// get or create resourceQuota
-	if err := r.syncResourceQuota(ctx, metering); err != nil {
+	} else if client.IgnoreNotFound(err) != nil {
 		return ctrl.Result{}, err
 	}
 
-	r.Logger.Info("update BillingList")
-	err := r.updateBillingList(ctx, &metering)
-	if err != nil {
-		r.Logger.Error(err, err.Error())
-		return ctrl.Result{}, err
+	var metering meteringv1.Metering
+	if err := r.Get(ctx, req.NamespacedName, &metering); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-	// Ensure the deployment size is the same as the spec
-	return ctrl.Result{Requeue: true, RequeueAfter: time.Minute}, nil
-}
 
-func (r *MeteringReconcile) createMetering(ctx context.Context, owner string, ns corev1.Namespace) (meteringv1.Metering, error) {
-	var metering meteringv1.Metering
-	err := r.Get(ctx, client.ObjectKey{Namespace: SealosSystemNamespace, Name: fmt.Sprintf("metering-%v", ns.Name)}, &metering)
-	if errors.IsNotFound(err) {
-		r.Logger.Info("creat metering")
-		metering = meteringv1.Metering{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      fmt.Sprintf("metering-%v", ns.Name),
-				Namespace: SealosSystemNamespace,
-			},
-			Spec: meteringv1.MeteringSpec{
-				Namespace: ns.Name,
-				Owner:     owner,
-				Resources: DefaultResourceQuota(),
-			},
-			Status: meteringv1.MeteringStatus{
-				TotalAmount:      0,
-				LatestUpdateTime: time.Now().Unix(),
-			},
+	if time.Now().Unix()-metering.Status.LatestUpdateTime >= int64(time.Minute.Seconds())*int64(metering.Spec.TimeInterval) {
```

Comment:
Add comment for this, split to a check func. IMO use second as TimeInterval is batter.

Reply:
no need to used second，usually will 1minute or 60minute ，not used often like 90s .